Example/Tests/MedStickerEncryptionTests.swift

```diff
@@ -174,4 +174,40 @@ final class MedStickerEncryptionTests: XCTestCase {
     // then
     XCTAssertTrue(signature.hasPrefix("adam-sha256"))
   }
+
+  func testGenerateFingerprintSecret_charlie__shouldGenerate132StringsAndContainVersionCharlie() {
+    let expectedLength = 64 * 2 + 8 // 64 bytes * 2(as hex string) + "charlie:" (8)
+
+    let fakePin = "fakePin"
+    let subject = try! MedStickerEncryption.generateFingerprintSecret(withPin: fakePin)
+
+    XCTAssertEqual(subject.count, expectedLength)
+    XCTAssertTrue(subject.hasPrefix("charlie:"))
+  }
+
+  func testEncrypAndDecrypt_charlie__shouldReturnSameData() {
+    let expectedData = "MyVeryImportantData".data(using: .utf8)!
+
+    let fakePin = "someRandomPin12345678"
+    let fakeBackendSecret = "someRandomBackendSecret"
+    let fakeSecondSalt = "someRandomSecondSalt"
+
+    let iv = randomData(count: 16)
+    let encrypted = try! MedStickerEncryption.encrypt(pin: fakePin, secret: fakeBackendSecret, salt: fakeSecondSalt, iv: iv, data: expectedData)
+
+    let decrypted = try! MedStickerEncryption.decrypt(pin: fakePin, secret: fakeBackendSecret, salt: fakeSecondSalt, iv: encrypted.attr.iv, data: encrypted.data)
+
+    XCTAssertEqual(decrypted, expectedData)
+  }
+
+  private func randomData(count: Int) -> Data {
```

Comment:
M just questioning if we really need this one here. At least would be nicety have it simplified a bit since you don't need to check for the errors and status.